### ISSUE TITLE
Server error when using the web based upload form

### ISSUE THREAD

**Issue by ajaycj1** [PlasmaFAIR/TDotDat#8]
It shows 'Server error' when using the web based upload form for a GENE linear run. I uploaded the standard parameter and nrg text files.

**Comment by ZedThree** [comment]
Hi @ajaycj1, finally found time to look at this. I've tested this locally using `parameters_0001` and `nrg_0001` from the TDotDat shared drive (let me know if you're using something very different!), and it looks like pyrokinetics assumes `minor_r` will be present in the `geometry` namelist.

I have a fix for pyrokinetics, PR incoming

**Comment by ajaycj1** [comment]
Yes I am used the same files from TDotDat shared drive.
If I remember correctly, I had already solved the issue of minor_r in Pyrokinetics for circular geometry. Are you using the most up-to-date version of Pyrokinetics?
Also, it appears that 'server error' has something to do with network access of the server from outside. The 'Permission denied' error I get when I try to upload using the script also seems to suggest that.

**Comment by ZedThree** [comment]
Possibly not, I'll update the version on the server and let you know when that's done.

"Server error" usually means there was some internal error, rather than access, for example pyrokinetics raised an exception

**Comment by ajaycj1** [comment]
Still says "Internal server error" when trying to upload GENE files - the same ones in TDotDat google drive.

**Comment by ZedThree** [comment]
It looks like this is because the growth rate is coming out as NaN and this is being rejected. We should probably strip NaNs from the data first.

Could you tell me which files you were trying to upload so I can check this?

**Comment by ajaycj1** [comment]
I am uploading parameters_0001 as the input file and nrg_0001 and omega_0001 as the output files (fromTDotDat_Googledrive/GENE_lin_kyscan). Tried with only nrg_0001 as output file, and a few other combinations. Says internal server error all the time.

The growth rate is converged and output in omega_0001 by GENE. I don't know if Pyrokinetics is trying to calculate growth rates from the nrg file. Usually if GENE finds that the growth rate is converged, there should be no problem finding that converged growth rate from the time trace of any qunatity.

**Comment by ZedThree** [comment]
Ah, I've been trying with `field_0001.h5`. It looks like pyrokinetics doesn't know how to read the ASCII `omega` files. It looks like it's in the form `ky gamma omega`? 

`omega` has a sign difference with the frequency pyrokinetics calculates from `fields.h5` -- I guess GENE chooses the electron diagmanetic drift direction as positive? Something to flag in PK though I think.

I'll try and get some kind of useful error page working too -- "server error" is not helpful!

**Comment by ajaycj1** [comment]
Yes, omega file is in the form ky gamma omega.
No, GENE chooses ion diamagnetic direction as positive.

If field_0001.h5 is included in the output files, all files including omega and nrg files are successfully uploaded now. 

I am wondering what the best way to upload data is. Since the first priority is to save the data first, maybe we should ignore the errors in Pyrokinetics, and save the data in any case?

I don't think people will be very enthusiastic about uploading data in the first place and any errors at this stage might be further demotivating. Also, solving all problems relating to Pyrokinetics may not be realistic, I think it will be an ongoing process.

On the other side, a lot of junk might end up in the database. And I don't know the security side of things.

**Comment by ZedThree** [comment]
Yes, we could create an entry and save the uploaded files, even if we can't extract all of the data. We should then be able to re-parse files later if pyrokinetics gets updated.

The issue about uploading junk is a fair point -- we might want some process to be able to "verify" and/or delete entries.
